src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,119 @@
+package storage;
+
+import duke.DukeException;
+import task.TaskList;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.Scanner;
+
+public class Storage implements IStorage{
+    private final String saveFilePath;
+
+    /**
+     * Constructor for Storage.
+     * @param saveFileDirectory
+     * @param saveFileName
+     * @throws DukeException
+     */
+    public Storage(String saveFileDirectory, String saveFileName) {
+        this.saveFilePath = saveFileDirectory + saveFileName;
+        File fileDirectory = new File(saveFileDirectory);
+        if (!fileDirectory.exists()) {
+            fileDirectory.mkdir();
+        }
+
+        File saveFile = new File(this.saveFilePath);
+        if (!saveFile.exists()) {
+            try {
+                saveFile.createNewFile();
+            } catch (IOException e) {
+                System.out.println(e);
+            }
+        }
+    }
+
+    /**
+     * Read tasks from file.
+     * @return TaskList
+     * @throws DukeException
+     */
+    public TaskList readFile() throws DukeException {
+        TaskList taskList = new TaskList();
+        File file = new File(saveFilePath);
+
+        try {
+            Scanner sc = new Scanner(file);
+            while (sc.hasNext()) {
+                taskList.addTaskNoMessage(processLine(sc.nextLine()));
+            }
+        } catch (FileNotFoundException e) {
+            throw DukeException.DukeFileNotFoundException();
+        } catch (DukeException d) {
+            throw d;
+        }
+        return taskList;
+    }
+
+    /**
+     * Write tasks to the file.
+     * @param taskList
+     * @throws DukeException
```

Comment:
Perhaps your comments could give more detailed descriptions of the params and exceptions eg. conditions for throwing an error. I've noticed the same issue in several other places too.